.gitignore

```diff
@@ -83,6 +83,7 @@ Makefile
 cmake_install.cmake
 CMakeFiles
 CMakeCache.txt
+/cmake-build-debug
```

Comment:
Should we change it to `./cmake-build-debug`? And CMake can specifiy an arbitrary build directory. For example, the command `cmake -B build` will create a `build` directory to contain CMake build files.

Though it makes sense if `cmake-build-debug` is generated by a specific IDE by default. For example, Visual Studio creates `out/` directory as the build directory. If then, please add a comment to explain.